sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -872,6 +872,20 @@ case class TempResolvedColumn(
   final override val nodePatterns: Seq[TreePattern] = Seq(TEMP_RESOLVED_COLUMN)
 }
 
+/**
+<<<<<<< HEAD
+ * Represents a `withColumns` operation. This operator both replaces existing columns
+ * (based on the name), or appends new columns.
+ */
+case class UnresolvedWithColumns(
+    projectList: Seq[NamedExpression],
```

Comment:
is it guaranteed that the projectList is `Seq[Alias]`?

Reply:
It sort of is. I like `NamedExpression` a bit better here, because it also allows for MultiAlias and friends.

Reply:
but what we really need is `Seq[String, Expression]`. Not sure if we want to extend this operator more in the future.